src/main/groovy/jenkins/automation/utils/ScmUtils.groovy

```diff
@@ -24,8 +32,9 @@ class ScmUtils {
                         branch("*/tags/$version")
                     }
                     relativeTargetDir(repo.sub_directory)
+                    shallowClone(shallow)
```

Comment:
you can use Elvis operator and do something like
shallowClone(repo.shallow?:false)
and that should do it. 

?: ==> elvis ( one of my groovy favs)